src/test/java/com/firebolt/jdbc/resultset/FireboltResultSetTest.java

```diff
@@ -1058,15 +1060,47 @@ void shouldGetObjectsForNumericTypes() throws SQLException {
 		inputStream = getInputStreamWithNumericTypes();
 		resultSet = createResultSet(inputStream);
 		resultSet.next();
+		assertEquals((byte)1, resultSet.getObject(1, Byte.class));
+		assertEquals((short)1, resultSet.getObject(1, Short.class));
 		assertEquals(1, resultSet.getObject(1, Integer.class));
+		assertEquals(1L, resultSet.getObject(1, Long.class));
 		assertEquals(new BigInteger("1"), resultSet.getObject(1, BigInteger.class));
 		assertEquals(1, resultSet.getObject(1, Long.class));
+		assertEquals(1, resultSet.getObject(1, Float.class));
+		assertEquals(1, resultSet.getObject(1, Double.class));
+
+		// the number is too big
+		FireboltException e = assertThrows(FireboltException.class, () -> resultSet.getObject(2, Integer.class));
+		assertEquals(TYPE_TRANSFORMATION_ERROR, e.getType());
+		assertEquals(NumberFormatException.class, e.getCause().getClass());
+
 		assertEquals(30000000000L, resultSet.getObject(2, Long.class));
 		assertEquals(new BigInteger("30000000000"), resultSet.getObject(2, BigInteger.class));
+		assertEquals(30000000000f, resultSet.getObject(2, Float.class));
+		assertEquals(30000000000., resultSet.getObject(2, Double.class));
+
+		assertEquals((byte)1, resultSet.getObject(3, Byte.class));
+		assertEquals((short)1, resultSet.getObject(3, Short.class));
+		assertEquals(1, resultSet.getObject(3, Integer.class));
+		assertEquals(1, resultSet.getObject(3, Long.class));
 		assertEquals(1.23f, resultSet.getObject(3, Float.class));
 		assertEquals(new BigDecimal("1.23"), resultSet.getObject(3, BigDecimal.class));
+		assertEquals(1.23, resultSet.getObject(3, Double.class));
+
+		assertEquals((byte)1, resultSet.getObject(4, Byte.class));
+		assertEquals((short)1, resultSet.getObject(4, Short.class));
+		assertEquals(1, resultSet.getObject(4, Integer.class));
+		assertEquals(1, resultSet.getObject(4, Long.class));
+		assertEquals(1.23456789012f, resultSet.getObject(4, Float.class));
 		assertEquals(1.23456789012, resultSet.getObject(4, Double.class));
 		assertEquals(new BigDecimal("1.23456789012"), resultSet.getObject(4, BigDecimal.class));
+
+		//assertEquals((byte)1, resultSet.getObject(5, Byte.class));
+		//assertEquals((short)1231232, resultSet.getObject(5, Short.class));
```

Comment:
Why do we have these commented out? Should they be working?

Reply:
fixed test and the code. 10x